### PR TITLE
[Merged by Bors] - feat(StructureGroupoid): one small lemma and some docstrings

### PR DESCRIPTION
**Pull request by winstonyin**
I added an iff version of `mem_of_eqOnSource`. This is to mirror how lemmas like `HasDerivAt.congr_of_eventuallyEq` have an iff counterpart `Filter.EventuallyEq.hasDerivAt_iff`.

I also wrote the docstrings for fields of `StructureGroupoid`.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by grunweg** [comment]
FWIW, #9832 will update the `nolints.json` files (and, by propagating leanprover/std4#269, means all entries but for `members` are not required any more).

File: Mathlib/Geometry/Manifold/ChartedSpace.lean [hunk 1]
```diff
@@ -161,12 +161,19 @@ We use primes in the structure names as we will reformulate them below (without
 /-- A structure groupoid is a set of partial homeomorphisms of a topological space stable under
 composition and inverse. They appear in the definition of the smoothness class of a manifold. -/
 structure StructureGroupoid (H : Type u) [TopologicalSpace H] where
+  /-- Members of the structure groupoids are partial homeomorphisms. -/
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/ChartedSpace.lean:164]: I would use singular here.

File: Mathlib/Geometry/Manifold/ChartedSpace.lean [hunk 2]
```diff
@@ -161,12 +161,19 @@ We use primes in the structure names as we will reformulate them below (without
...
   members : Set (PartialHomeomorph H H)
+  /-- Structure groupoids are stable under composition. -/
   trans' : ∀ e e' : PartialHomeomorph H H, e ∈ members → e' ∈ members → e ≫ₕ e' ∈ members
+  /-- Structure groupoids are stable under inverse. -/
   symm' : ∀ e : PartialHomeomorph H H, e ∈ members → e.symm ∈ members
+  /-- The identity is in the structure groupoid. -/
   id_mem' : PartialHomeomorph.refl H ∈ members
+  /-- If the restriction of a partial homeomorphism to some open set around every point in its
```
Comment (grunweg) [inline Mathlib/Geometry/Manifold/ChartedSpace.lean:172]: I think this docstring hard to parse: is this better?
"Let `e` be a partial homeomorphism. If for every `x \in e.source`, the restriction of `e` to some open set around `x` lies in the groupoid, then `e` lies in the groupoid."

**Review by grunweg — APPROVED** [review]
Looks good to me. I suggested a few tweaks to the docstrings; I would approve with these taken into account.

File: Mathlib/Geometry/Manifold/ChartedSpace.lean [hunk 3]
```diff
@@ -252,6 +259,10 @@ theorem StructureGroupoid.mem_of_eqOnSource (G : StructureGroupoid H) {e e' : Pa
   G.mem_of_eqOnSource' e e' he h
 #align structure_groupoid.eq_on_source StructureGroupoid.mem_of_eqOnSource
 
+theorem StructureGroupoid.mem_iff_of_eqOnSource (G : StructureGroupoid H)
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/ChartedSpace.lean:262]:
```suggestion
theorem StructureGroupoid.mem_iff_of_eqOnSource {G : StructureGroupoid H}
```
When a variable can be inferred from any side of an iff, it should be implicit.

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

**Comment by grunweg** [comment]
I took the liberty to merge master and address the review comment; I would like to see this land :-)
@winstonyin @sgouezel Would one of you like to merge this?
